### ISSUE TITLE
Upgrade to Rails 8.0

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#12977]
## Description

Follow up from:

- [x] #11673
- [ ] #13290

Resources:

* Rails 8.0 requires Ruby 3.2.0 or newer.
* [An upgrade handbook to Rails 8 - Rails World 2024](https://www.youtube.com/watch?v=HTRNa-U_KKk)
* [Upgrading from Rails 7.2 to Rails 8.0](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-7-2-to-rails-8-0)
* [Ruby on Rails 8.0 Release Notes](https://guides.rubyonrails.org/8_0_release_notes.html)

**Comment by piyush828-design** [comment]
@mkllnk I can work on this.

**Comment by mkllnk** [comment]
Yes, you can. But we want to upgrade to Rails 7.1 first. We are pretty close. Would you like to tackle that first?

**Comment by piyush828-design** [comment]
Yes, sure.
